lib/rgeo/geographic/factory.rb

```diff
@@ -36,17 +36,25 @@ def initialize(impl_prefix, opts = {}) # :nodoc:
         @coordinate_dimension += 1 if @support_m
         @spatial_dimension = @support_z ? 3 : 2
 
-        @srid = (opts[:srid] || 4326).to_i
-        @proj4 = opts[:proj4]
-        if @proj4 && CoordSys.check!(:proj4)
-          if @proj4.is_a?(String) || @proj4.is_a?(Hash)
-            @proj4 = CoordSys::Proj4.create(@proj4)
-          end
-        end
         @coord_sys = opts[:coord_sys]
+        coord_sys_class = opts[:coord_sys_class]
+        unless coord_sys_class.is_a?(Class)
+          coord_sys_class = CoordSys::CONFIG.default_coord_sys_class
+        end
+
         if @coord_sys.is_a?(String)
-          @coord_sys = CoordSys::CS.create_from_wkt(@coord_sys)
+          @coord_sys = coord_sys_class.create_from_wkt(@coord_sys)
+        end
+
+        unless @coord_sys.is_a?(CoordSys::CS::CoordinateSystem)
+          # TODO: should we raise here?
+          @coord_sys = nil
+        end
```

Comment:
yeah I guess I can just keep it and if it's not a CoordinateSystem, it will end up raising down the line when you try to invoke a transformation on it